src/neo-vm/Types/PrimitiveType.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Numerics;
+
+namespace Neo.VM.Types
+{
+    public abstract class PrimitiveType : StackItem
+    {
+        public override bool Equals(StackItem other)
+        {
+            if (ReferenceEquals(this, other)) return true;
+            if (other is null) return false;
```

Comment:
`ReferenceEquals` and null check is made twice, for example in `Boolean` I think that we should remove this two checks in this `base.Equals`

Reply:
It can't be removed. Please check the class `ByteArray`.